### PR TITLE
First design of popularity correction

### LINKED ISSUE #161: Add logits correction for ItemsPredictionSampled (sampled softmax)
The `ItemsPredictionSampled` (previous SampledSoftmax) is doing popularity-based sampling but is missing the logits correction according to the items popularity. This is a pending issue from the PR  https://github.com/NVIDIA-Merlin/models/pull/152

### PR DESCRIPTION
**Pull request by sararb**
First implementation of popularity bias correction. 

- [x] First design of popularity sampling correction that uses a fixed frequency table
- [x] Store label ids for positive and negatives in the ItemRetrievalScorer block for the frequency table lookup 
- [x] Link  popularity sampling block to the Item Retrieval task

Closes #161

### CODE REVIEW

File: merlin/models/tf/blocks/core/transformations.py [hunk 1]
```diff
@@ -556,3 +556,76 @@ def call_outputs(
...
+
+    TODO: Use the schema to infer the path to the item frequency parquet table
+
+    Parameters:
+    ----------
+    item_frequency : Union[tf.Tensor]
+        The item frequency table
+    """
+
+    def __init__(self, item_frequency: Union[tf.Tensor, str], **kwargs):
```
Comment (gabrielspmoreira) [inline merlin/models/tf/blocks/core/transformations.py:576]: I think `item_frequency` should always be a tf.Tensor (not str), don't?

Comment (gabrielspmoreira) [inline merlin/models/tf/blocks/core/transformations.py:576]: Maybe we should also have the schema as an argument, so that we retrieve the item_id cardinality from the schema and check if it matches the shape of `item_frequency`, just for defensive programming.

File: merlin/models/tf/blocks/core/transformations.py [hunk 2]
```diff
@@ -556,3 +556,76 @@ def call_outputs(
...
+        super().__init__(**kwargs)
+        candidate_probabilities = tf.cast(
+            item_frequency / tf.reduce_sum(item_frequency), tf.float32
+        )
+        self.candidate_probabilities = tf.Variable(
```
Comment (gabrielspmoreira) [inline merlin/models/tf/blocks/core/transformations.py:581]: I think we should also have an `update()` method, in case we need to update the item popularities when fine tuning a pre-trained model

**Comment by gabrielspmoreira** [comment]
We need to think how we can access the `PopularityLogitsCorrection.from_parquet()` from the higher-level building blocks (e.g. from `ItemRetrievalTask`)

File: merlin/models/tf/blocks/core/transformations.py (hunk 1 above)
Comment (sararb) [reply merlin/models/tf/blocks/core/transformations.py]: yes, I was testing different designs and forgot to remove the `str` type. As this is handled by the `from_parquet` method in the current implementation draft.

**Review by gabrielspmoreira — APPROVED** [review]
Sounds good to me! Just pushed a commit with some naming and documentation improvements, and added some checks and tests
